pkgs/by-name/gc/gclient2nix/gclient2nix.py

```diff
@@ -0,0 +1,226 @@
+
+        prefix = f"{path}/" if evaluated.get("use_relative_paths", False) else ""
+
+        self.deps = {
+            prefix + dep_name: repo_from_dep(dep)
+            for dep_name, dep in evaluated.get("deps", {}).items()
+            if (
+                gclient_eval.EvaluateCondition(dep["condition"], repo_vars)
+                if "condition" in dep
+                else True
+            )
+            and repo_from_dep(dep) != None
+        }
+
+        for key in evaluated.get("recursedeps", []):
+            dep_path = prefix + key
+            if dep_path in self.deps:
+                self.deps[dep_path].get_deps(repo_vars, dep_path)
+
+    def eval(self) -> None:
+        self.get_deps(
+            {
+                **{
+                f"checkout_{platform}": platform == "linux"
+                for platform in ["ios", "chromeos", "android", "mac", "win", "linux"]
+                },
+                **{
+                f"checkout_{arch}": True
+                for arch in ["x64", "arm64", "arm", "x86", "mips", "mips64", "ppc"]
+                },
+            },
+            "",
+        )
+
+    def prefetch(self) -> None:
+        self.hash = get_repo_hash(self.fetcher, self.args)
+
+    def prefetch_all(self) -> int:
+        return sum(
+            [dep.prefetch_all() for [_, dep] in self.deps.items()],
+            [delayed(self.prefetch)()],
+        )
+
+    def flatten_repr(self) -> dict:
+        return {"fetcher": self.fetcher, "attrs": {**({"hash": self.hash} if hasattr(self, "hash") else {}), **self.args}}
+
+    def flatten(self, path: str) -> dict:
+        out = {path: self.flatten_repr()}
+        for dep_path, dep in self.deps.items():
+            out |= dep.flatten(dep_path)
+        return out
+
+    def get_file(self, filepath: str) -> str:
+        raise NotImplementedError
+
+
+class GitRepo(Repo):
+    def __init__(self, url: str, rev: str) -> None:
+        super().__init__()
+        self.fetcher = "fetchgit"
+        self.args = {
+            "url": url,
+            "rev": rev,
+        }
+
+
+class GitHubRepo(Repo):
+    def __init__(self, owner: str, repo: str, rev: str) -> None:
+        super().__init__()
+        self.fetcher = "fetchFromGitHub"
+        self.args = {
+            "owner": owner,
+            "repo": repo,
+            "rev": rev,
+        }
+
+    def get_file(self, filepath: str) -> str:
+        return (
+            urlopen(
+                f"https://raw.githubusercontent.com/{self.args['owner']}/{self.args['repo']}/{self.args['rev']}/{filepath}"
+            )
+            .read()
+            .decode("utf-8")
+        )
+
+
+class GitilesRepo(Repo):
+    def __init__(self, url: str, rev: str) -> None:
+        super().__init__()
+        self.fetcher = "fetchFromGitiles"
+        self.args = {
+            "url": url,
+            "rev": rev,
+        }
+
+        if url == "https://chromium.googlesource.com/chromium/src.git":
```

Comment:
Add a HACK: comment that this is to trim sources for the chromium source derivation to not get past the hydra output size limit.